main.tf

```diff
@@ -15,20 +15,20 @@ locals {
   atlantis_url_events = "${local.atlantis_url}/events"
 
   # Include only one group of secrets - for github, gitlab or bitbucket
-  has_secrets = var.atlantis_gitlab_user_token != "" || var.atlantis_github_user_token != "" || var.atlantis_bitbucket_user_token != ""
+  has_secrets = var.atlantis_gitlab_user_token != "" || var.atlantis_github_user_token != "" || var.atlantis_bitbucket_user_token != "" || var.atlantis_azuredevops_user_token != ""
 
   # token
-  secret_name_key        = local.has_secrets ? var.atlantis_gitlab_user_token != "" ? "ATLANTIS_GITLAB_TOKEN" : var.atlantis_github_user_token != "" ? "ATLANTIS_GH_TOKEN" : "ATLANTIS_BITBUCKET_TOKEN" : ""
-  secret_name_value_from = local.has_secrets ? var.atlantis_gitlab_user_token != "" ? var.atlantis_gitlab_user_token_ssm_parameter_name : var.atlantis_github_user_token != "" ? var.atlantis_github_user_token_ssm_parameter_name : var.atlantis_bitbucket_user_token_ssm_parameter_name : ""
+  secret_name_key        = local.has_secrets ? var.atlantis_gitlab_user_token != "" ? "ATLANTIS_GITLAB_TOKEN" : var.atlantis_github_user_token != "" ? "ATLANTIS_GH_TOKEN" : var.atlantis_bitbucket_user_token != "" ? "ATLANTIS_BITBUCKET_TOKEN" : "ATLANTIS_AZUREDEVOPS_TOKEN" : ""
+  secret_name_value_from = local.has_secrets ? var.atlantis_gitlab_user_token != "" ? var.atlantis_gitlab_user_token_ssm_parameter_name : var.atlantis_github_user_token != "" ? var.atlantis_github_user_token_ssm_parameter_name : var.atlantis_bitbucket_user_token != "" ? var.atlantis_bitbucket_user_token_ssm_parameter_name : var.atlantis_azuredevops_user_token_ssm_parameter_name : ""
 
   # webhook
-  secret_webhook_key = local.has_secrets || var.atlantis_github_webhook_secret != "" ? var.atlantis_gitlab_user_token != "" ? "ATLANTIS_GITLAB_WEBHOOK_SECRET" : var.atlantis_github_user_token != "" || var.atlantis_github_webhook_secret != "" ? "ATLANTIS_GH_WEBHOOK_SECRET" : "ATLANTIS_BITBUCKET_WEBHOOK_SECRET" : ""
+  secret_webhook_key = local.has_secrets || var.atlantis_github_webhook_secret != "" ? var.atlantis_gitlab_user_token != "" ? "ATLANTIS_GITLAB_WEBHOOK_SECRET" : var.atlantis_github_user_token != "" || var.atlantis_github_webhook_secret != "" ? "ATLANTIS_GH_WEBHOOK_SECRET" : var.atlantis_bitbucket_user_token != "" ? "ATLANTIS_BITBUCKET_WEBHOOK_SECRET" : "ATLANTIS_AZUREDEVOPS_WEBHOOK_PASSWORD" : ""
```

Comment:
This is insanely hard to read, my suggestion would be to write it as follows:
```tf
github_secret_name = var.atlantis_github_webhook_secret != "" ? "ATLANTIS_GH_WEBHOOK_SECRET" : ""
gitlab_secret_name = var.atlantis_gitlab_user_token != "" ? "ATLANTIS_GITLAB_WEBHOOK_SECRET" : ""
bitbucket_secret_name = var.atlantis_bitbucket_user_token != "" ? "ATLANTIS_BITBUCKET_WEBHOOK_SECRET" : ""
azuredevops_secret_name = var.atlantis_azuredevops_user_token != "" ? "ATLANTIS_AZUREDEVOPS_WEBHOOK_PASSWORD": ""

secret_webhook_key = try(coalesce(github_secret_name, gitlab_secret_name, bitbucket_secret_name, azuredevops_secret_name), "")
```

Reply:
agreed. this can also benefit from a single coalesce statement with multiple arguments